src/synapse-core/Actions/RunnerAction.vala

```diff
@@ -21,6 +22,17 @@
 */
 
 private class Synapse.RunnerAction: Synapse.BaseAction {
+
+    private Slingshot.Backend.SwitcherooControl switcheroo_control;
+
+    construct {
+        try {
```

Comment:
Same comment as before, I don't think this constructor can throw exceptions, so this try/catch probably not needed.